File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/services/v2/checker/CheckerServiceFactoryService.kt
package de.thm.ii.fbs.services.v2.checker

import de.thm.ii.fbs.services.v2.checker.trait.CheckerService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Service
@Service
class CheckerServiceFactoryService {
    @Autowired
    lateinit var applicationContext: ApplicationContext

    /**
     * Gets a checker for the given service
     * @param service the service to get a checker for
     * @return the checker
     */
    fun apply(service: String): CheckerService {
        return when(service)  {
            "spreadsheet" -> applicationContext.getBean(classOf[SpreadsheetCheckerService])
            "excel" -> applicationContext.getBean(classOf[ExcelCheckerService])
            "sql-checker" -> applicationContext.getBean(classOf[SqlCheckerRemoteCheckerService])
            else -> applicationContext.getBean(classOf[RemoteCheckerService])
        }
    }
}

File: modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/services/checker/CheckerServiceFactoryService.scala
package de.thm.ii.fbs.services.checker

import de.thm.ii.fbs.services.checker.`trait`.CheckerService
import de.thm.ii.fbs.services.checker.excel.ExcelCheckerService
import de.thm.ii.fbs.services.checker.math.SpreadsheetCheckerService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Service

/**
  * A Factory for Checker Services
  */
@Service
class CheckerServiceFactoryService {
  @Autowired
  private val applicationContext: ApplicationContext = null

  /**
    * Gets a checker for the given service
    * @param service the service to get a checker for
    * @return the checker
    */
  def apply(service: String): CheckerService = service match {
    case "spreadsheet" => applicationContext.getBean(classOf[SpreadsheetCheckerService])
    case "excel" => applicationContext.getBean(classOf[ExcelCheckerService])
    case "sql-checker" => applicationContext.getBean(classOf[LegacySqlCheckerService])
    case "ai-supported-sql-analyser" => applicationContext.getBean(classOf[AiSupportedSqlAnalysisService])
    case _: String => applicationContext.getBean(classOf[RemoteCheckerService])
  }
}
